sources/kernel/interrupts.c

```diff
@@ -0,0 +1,210 @@
+
+#include <brutal/alloc.h>
+#include <brutal/base.h>
+#include <brutal/ds.h>
+#include <brutal/log.h>
+#include <syscalls/types.h>
+#include "kernel/interrupts.h"
+#include "kernel/task.h"
+
+static IrqBinding *interrupts = nullptr;
+static size_t interrupt_count = 0;
```

Comment:
`interrupt_count` is that variable really needed?

Reply:
yeah it simplifies some case where we delete every interrupts...